tensorflow/stream_executor/cuda/cuda_dnn.cc

```diff
@@ -1514,9 +1517,15 @@ class CudnnRnnSequenceTensorDescriptor
         /*batchSize=*/batch_size, /*vectorSize=*/data_size,
         /*seqLengthArray=*/seq_lengths_array,
         /*paddingFill*/ (void*)&padding_fill));
-    return CudnnRnnSequenceTensorDescriptor(
+    auto rnn_seq_tensor = CudnnRnnSequenceTensorDescriptor(
         parent, max_seq_length, batch_size, data_size, data_type,
         std::move(data_desc), std::move(tensor_desc));
+#if CUDNN_VERSION >= 8000
+    // Keep a copy of seq_lengths in the object to make it live long enough
```

Comment:
What guarantees that `rnn_seq_tensor` will stay alive until the H2D copy is complete?  Please add a comment to the `CopyRnnSeqLengthsH2D` function since IMO that's a natural questions folks later reading this code will have.

Reply:
I don't think this is guaranteed. I think it still might be destroyed before the H2D copy occurs.

I reverted this commit and add a BlockHostUntilDone call after the H2D copy to enforce the execution.